### ISSUE TITLE
Drag and Drop file import

### ISSUE THREAD

**Issue by CashWilliams** [OscarGodson/EpicEditor#157]
It would be a nice feature to allow a user to 'drag & drop' a file from their local file system onto the editor. This would then open the file and import the contents into the editor. Something like editor.open() or editor.importFile().

Should only work with markdown files, so probably .md and .txt.

Also, this drag and drop feature may only be a feature of Chrome.

**Comment by dadambickford** [comment]
I believe the jQuery UI has drag and drop methods, but i'm not sure how to
implent them here, that would be up to Oscar, but I think it sounds cool.

On Thu, Jul 19, 2012 at 9:28 AM, Cash Williams <
reply@reply.github.com

**Comment by OscarGodson** [comment]
@dadambickford I think he's talking more about dragging an actual file into the browser window, dropping it, and the content appearing in the editor not moving elements around the page.

**Comment by jpdevries** [comment]
I've had this idea, but less for text importing and more to support uploading images. I'm not sure if EpicEditor aims to do something like that (support image uploads if provided a connector to upload to)

**Comment by EranSch** [comment]
+1 @jpdevries - Such a feature would make an excellent complement to EE. Especially in conjunction with a configurable connector which to allow storage of the site on the server or even a third party host.

The functionality of Github's markdown editor is exactly what I envision. The user can drag an image into the editor and the image upload is handled silently with only the relevant markdown inserted into the editor...

`![alt](url-to-uploaded-image)`

Very cool prospect.

**Comment by Ryel** [comment]
@OscarGodson  Is this something that aligns with the EE roadmap?

**Comment by OscarGodson** [comment]
@ryel indeed, but it should be an extension. Trying to keep the core editor to just text editing. But, extensions should be super easy to install like just including a script tag :)
